src/Microsoft.Framework.Configuration/Internal/ConfigurationFocus.cs

```diff
@@ -6,77 +6,95 @@
 
 namespace Microsoft.Framework.Configuration.Internal
 {
-    public class ConfigurationFocus : IConfiguration
+    public class ConfigurationFocus : IConfigurationSection
     {
+        private string _key { get; set; }
+        private string _value { get; set; }
+
         private readonly string _prefix;
-        private readonly IConfiguration _root;
+        private readonly IConfigurationSection _root;
 
-        public ConfigurationFocus(IConfiguration root, string prefix)
+        public string Key
```

Comment:
Will stop looking and this class as it needs to go away/be collapsed into `ConfigurationSection` and most of the functionality lifted to a base Configuration class.